TypeScript (#1819)
enkidevs/curriculum: typescript/introduction/intro-typescript/how-can-typescript-help-me.md

---
author: kapnobatai136

type: normal

category: must-know

aspects:
  - introduction

---

# How Can TypeScript Help Me

---
## Content

Knowing that TypeScript is nothing more than JavaScript with extra features, you're ready to find out how you can benefit from using TS.

> One of the main features, as its name suggest, is that it allows a lot more control over types. By type checking the code, you can avoid many common bugs and increase the efficiency of your development process.
> 
> Interfaces are another feature introduced by TypeScript, and these allow you to specify a structure of types that can be used as arguments to a function, or simply as an expected object.
> 
> Here are some other features or benefits you might gain from using TypeScript:
> - makes code easier to read and understand
> - supports the latest browsers
> - supports static typing
> - supports generics
> - supports modules

Reviewer: ```suggestion
The main feature, as its name suggest, is that it allows the usage of static types. 
The static in this context means that types are checked by looking at the written code without needing to run it. 

In other words, TypeScript checks your code by making sure you use the same names and shapes where you said you would.

You can think of it as a spellchecker for JavaScript.

This is useful because making sure that the same types of data are used throughout the code can help avoid bugs. It makes the code consistent, and less prone to typos.

It also makes the development faster because you can catch problems early on. 
Since the code doesn't need to run to be type-checked (remember, static types), your editor can integrate with TypeScript and underline in red any type mismatches in the code as you're writing it. This means you write safer code from the start.

Since TypeScript is introduced as another layer on top of JavaScript, it also allows us to use the latest JavaScript features and handles the conversion for old platforms behind the scenes. 
```

---
## Practice

> TypeScript gives you ??? control over types.
> 
> * more
> * less
> * the same

Reviewer: ```suggestion
TypeScript requires your code to run and produce an output before it can check for types.

* No
* Yes
```

---
## Revision

> Which of the following is not a benefit of using TypeScript?
> 
> ???
> 
> * increased bug creation chance
> * increased readability
> * supports type checking
> * supports generics

Reviewer: ```suggestion
TypeScript uses a ??? type system.

???

* static
* dynamic
* runtime
* duck
```
